core/store/migrate/migrations/0171_add_s4_functions_table.sql

```diff
@@ -0,0 +1,31 @@
+-- +goose Up
+
+CREATE SCHEMA "s4";
+
+CREATE TABLE "s4".functions(
+    id BIGSERIAL PRIMARY KEY,
+    address NUMERIC(78,0) NOT NULL,
+    slot_id INT NOT NULL,
+    version INT NOT NULL,
+    expiration BIGINT NOT NULL,
+    confirmed BOOLEAN NOT NULL,
+    payload BYTEA NOT NULL,
+    signature BYTEA NOT NULL,
+    updated_at TIMESTAMPTZ NOT NULL
+);
+
+CREATE UNIQUE INDEX functions_address_slot_id_idx ON "s4".functions(address, slot_id);
+CREATE INDEX functions_version_idx ON "s4".functions(version);
```

Comment:
Does this make sense? Do you ever do global lookups by version?

Reply:
I thought it is good for Update, but looks like you are right, this is not needed for that context.